samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
+    try {
+      String content = "plan='" + planJson + "'";
+      String binPath = System.getenv(ShellCommandConfig.JOB_BIN_DIR());
```

Comment:
I updated the patch with this change. Now the plan.json will be under /plan folder. I tested with the samza-api-test job and it works well. Thanks a lot for the suggestion!